docs/source/performance/cache-backends.mdx

```diff
@@ -0,0 +1,81 @@
+
+Apollo Server has several features that take advantage of a cache backend (e.g., [automatic persisted queries](./performance/apq), the [response cache plugin](./performance/caching#caching-with-responsecacheplugin-advanced), and [`RESTDataSource`](./data/data-sources#restdatasource-reference)). By default, Apollo Server uses a cache in memory, but you can configure it to use a different backend such as Redis or Memcached.
+
+You can specify a cache backend by passing a `cache` option to the `ApolloServer` constructor. Your specified cache backend _must_ implement the [`KeyValueCache`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyValueCache#keyvaluecache-interface) interface from the `@apollo/utils.keyvaluecache` package.
+
+There are many cache backend implementations to choose from, including several implementations wrapped by Apollo. For example, Apollo maintains an implementation of `InMemoryLRUCache` in the `@apollo/utils.keyvaluecache` package. Apollo also provides a wrapper class for the  [`keyv` package](https://www.npmjs.com/package/keyv) (which implements several cache backends) named [`KeyvAdapter`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyvAdapter#keyvadapter-class) in the `@apollo/utils.keyvadapter` package.
+
+Unfortunately, the default configuration of Apollo Server exposes you to denial of service attacks.
+This is because APQs are enabled by default and the default cache in Apollo Server 3 is unbounded, meaning an attacker can exhaust your memory and crash your server. The default cache in Apollo Server 4 will be bounded; **we recommend opting in to that behavior by providing the `cache: "bounded"` option to your Apollo Server constructor or configuring the `cache` yourself.**
+
+## Configuring in-memory caching
+
+**Apollo Server's default caching features use an unbounded cache, which is not safe for production use.** If you'd like to configure the in-memory cache, Apollo provides the [`InMemoryLRUCache`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyValueCache#inmemorylrucache) class from the `@apollo/utils.keyvaluecache` package.
+
+The `InMemoryLRUCache` class is a wrapper around the `lru-cache` package and has a default maximum of approximately 30MiB of memory. You can configure an instance of `InMemoryLRUCache` with the same options as found in the `lru-cache` package, see the [`lru-cache` documentation](https://www.npmjs.com/package/lru-cache) for more details. You can use Apollo's `InMemoryLRUCache` by passing it to the `cache` option of the `ApolloServer` constructor like so:
+
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // Note: this is equivalent to providing `cache: "bounded"`
+  cache: new InMemoryLRUCache(),
+});
+```
+
```

Comment:
Maybe add a paragraph saying something like

As a shorthand, you can write `cache: "bounded"`; this is equivalent to `cache: new InMemoryLRUCache()` and doesn't require you to `npm install @apollo/utils.keyvaluecache` yourself.